ownership.go

```diff
@@ -33,46 +33,50 @@ type Ownership struct {
 }
 
 // NewOwnership initializes default file ownership struct.
-func NewOwnership(username, groupname string) Ownership {
-	return Ownership{
-		UID: lookupUID(username),
-		GID: lookupGID(groupname),
+func NewOwnership(username, groupname string) (Ownership, error) {
+	uid, err := lookupUID(username)
+	if err != nil {
+		return Ownership{}, err
 	}
+	gid, err := lookupGID(groupname)
+	if err != nil {
+		return Ownership{}, err
+	}
+	return Ownership{
+		UID: uid,
+		GID: gid,
+	}, nil
 }
 
-// lookupUID resolves a username to a numeric id. Current euid is returned on failure.
-func lookupUID(username string) uint32 {
+// lookupUID resolves a username to a numeric id.
+func lookupUID(username string) (uint32, error) {
 	u, err := user.Lookup(username)
 	if err != nil {
-		log.Printf("Error resolving uid for %v: %v\n", username, err)
-		return uint32(os.Geteuid())
+		return 0, fmt.Errorf("Error resolving uid for %v: %v\n", username, err)
 	}
 
 	uid, err := strconv.ParseUint(u.Uid, 10 /* base */, 32 /* bits */)
 	if err != nil {
-		log.Printf("Error resolving uid for %v: %v\n", username, err)
-		return uint32(os.Geteuid())
+		return 0, fmt.Errorf("Error resolving uid for %v: %v\n", username, err)
```

Comment:
Do we want to make this a slightly different error message than above? It looks like this is just due to failing to parse as opposed to doing a lookup.